packages/netlify-cms-core/src/backend.ts

```diff
@@ -551,20 +482,24 @@ export class Backend {
 
     const integration = selectIntegration(state.integrations, null, 'assetStore');
 
-    const [loadedEntry, mediaFiles] = await Promise.all([
-      this.implementation.getEntry(path),
-      collection.has('media_folder') && !integration
-        ? this.implementation.getMedia(selectMediaFolder(state.config, collection, path))
-        : Promise.resolve(state.mediaLibrary.get('files') || []),
-    ]);
+    const loadedEntry = await this.implementation.getEntry(path);
 
     const entry = createEntry(collection.get('name'), slug, loadedEntry.file.path, {
       raw: loadedEntry.data,
       label,
-      mediaFiles,
+      mediaFiles: [],
     });
 
-    return this.entryWithFormat(collection)(entry);
+    const entryWithFormat = this.entryWithFormat(collection)(entry);
```

Comment:
need to parse the entry before retrieving the media files since we need the entry data to calculate the media folder template.